csfieldguide/static/interactives/regular-expression-filter/js/regular-expression-filter.js

```diff
@@ -25,32 +29,59 @@ function processRegularExpression() {
   var regex = document.getElementById('interactive-regular-expression-filter-regex').value
   RegularExpressionFilter.regex = new RegExp(regex);
   displayWords();
-  updateWordCount();
 };
 
 
 // Displays the filtered words
 function displayWords() {
+  // Get the button out of the way so it doesn't get deleted
+  $('#interactive-regular-expression-filter-load-more').appendTo($('#interactive-regular-expression-filter-loading'));
+
+  var window = document.getElementById('interactive-regular-expression-filter-words');
+  window.innerHTML = "";
   RegularExpressionFilter.displayed = 0;
-  html = ""
-  var display = document.getElementById('interactive-regular-expression-filter-words');
-  for (var i = 0; i < RegularExpressionFilter.words.length; i++) {
-    if (RegularExpressionFilter.regex.test(RegularExpressionFilter.words[i])) {
-      html += RegularExpressionFilter.words[i] + '\n';
-      RegularExpressionFilter.displayed++;
+  RegularExpressionFilter.matched = 0;
+  MatchedWords = [];
+  var length = RegularExpressionFilter.words.length;
+  var words = RegularExpressionFilter.words;
+  for (var i = 0; i < length; i++) {
+    if (RegularExpressionFilter.regex.test(words[i])) {
+      MatchedWords.push(words[i]);
+      RegularExpressionFilter.matched++;
     }
   }
-  display.innerHTML = html.trim();
+  updateWordCount();
+  displayMoreWords(5000);
 };
 
+// Displays 1000 more words (unless val is specified) and moves the 'display more' button to the end of the list
+function displayMoreWords(val) {
+  var num = 1000;
```

Comment:
I think we could probably load 5000 words each time. Then you wouldn't need to pass in any parameters and the user doesn't have to click 'load more' as often.

Reply:
I'll change it but there's no reason to click load more in the first place. The interactive is about pattern matching; if they want the user can just specify to match words further down the alphabet. Also the more words that are displayed the longer it takes to display more words - up to a second per thousand in chrome towards the end